src/libmongoc/src/mongoc/mongoc-stream-tls.c

```diff
@@ -255,13 +255,13 @@ mongoc_stream_tls_new_with_hostname_and_openssl_context (
 
    /* !client is only used for testing,
     * when the streams are pretending to be the server */
-   if (!client || opt->weak_cert_validation) {
+   if (opt && (!client || opt->weak_cert_validation)) {
```

Comment:
Should this check `opt`, or should it assert `opt`?

Reply:
Yeah, it seems that an assert works here. I have a clear recollection of attempting to use `BSON_ASSERT` and it triggering a bunch of failures, but I can't find it in my history of patch builds. I attempted another patch build with `BSON_ASSERT` rather than the Boolean check and that seems to work. Commit incoming.